gene_transformer/blast.py

```diff
@@ -3,94 +3,87 @@
+        temp_csv_dir: Path,
+    ) -> None:
         self.database_file = database_file
 
-        self.temp_fasta_dir = temp_fasta_dir
-        self.temp_fasta = Path(
-            str(temp_fasta_dir) + "/test_seq{}.fasta".format(hash(sequence))
-        )
-        # need to save sequence as fasta file in order to run Blast
+        self.temp_fasta = temp_fasta_dir / f"test_seq{hash(sequence)}.fasta"
+        # Need to save sequence as fasta file in order to run Blast
         make_fasta_from_seq(sequence, self.temp_fasta)
 
-        self.temp_csv_dir = temp_csv_dir
-        self.temp_csv = Path(
-            str(temp_csv_dir) + "/blast_res{}.csv".format(hash(sequence))
-        )
+        self.temp_csv = temp_csv_dir / f"blast_res{hash(sequence)}.csv"
 
-        # cannot get scores until blast has been run
+        # Cannot get scores until blast has been run
         self.ran_blast = False
 
-        self.scores = []
-        self.top_score = None
+        # TODO: Is this the correct type?
+        self.scores: List[float] = []
+
+    def _warning_message(self) -> None:
+        print(
+            "Scores have not yet been defined. Make sure"
+            " you've called run_blast() and get_scores()."
+        )
 
-    def run_blast(self):
+    def run_blast(self) -> None:
         # run local blastn given parameters in init, REQUIRES LOCAL INSTALLATION OF BLAST
         command = "blastn -query {} -subject {} -out {} -outfmt 10".format(
             self.temp_fasta, self.database_file, self.temp_csv
         )
         subprocess.run(command, shell=True)
         self.ran_blast = True
 
-    def get_scores(self):
+    def get_scores(self) -> Optional[List[float]]:
+        if not self.ran_blast:
+            print("Blast has not yet been run. Call the run_blast() method first.")
+            return None
         try:
-            assert self.ran_blast
-            # read in csv where blast results were stored
+            # Read in csv where blast results were stored
             df = pd.read_csv(self.temp_csv, header=None)
-            # take column which specifies scores
+            # Take column which specifies scores
             self.scores = df[11].tolist()
             return self.scores
-        except AssertionError:
-            print("Blast has not yet been run. Call the run_blast() method first.")
-            return None
         except pd.errors.EmptyDataError:
             self.scores = [-1]
+            return None
 
-    def get_top_score(self):
-        try:
-            assert self.scores
+    def get_top_score(self) -> Optional[float]:
+        if self.scores:
             return self.scores[0]
-        except AssertionError:
-            print(
-                "Scores have not yet been defined. Make sure you've called run_blast() and get_scores()."
-            )
-            return None
 
-    def get_mean_score(self):
-        try:
-            assert self.scores
+        self._warning_message()
```

Comment:
Should we have it return None with a warning? It might be better to have it raise an Exception if this is called before `run_blast()` since this means the code is not being used properly. Otherwise we might get silent errors. @maxzvyagin

Reply:
Instead, we can simply run blast and get scores if they have not been run.